external/fv3fit/fv3fit/sklearn/transformer.py

```diff
@@ -0,0 +1,95 @@
+import joblib
+import numpy as np
+import os
+from sklearn.preprocessing import StandardScaler
+from sklearn.base import TransformerMixin
+from typing import Sequence
+import yaml
+
+from fv3fit._shared.predictor import Reloadable
+from fv3fit._shared import (
+    get_dir,
+    put_dir,
+)
+from fv3fit._shared import io
+
+
+def _ensure_sample_dim(x: np.ndarray) -> np.ndarray:
+    if x.ndim == 1:
+        return x.reshape(1, -1)
+    else:
+        return x
+
+
+@io.register("sk-transformer")
+class SkTransformer(Reloadable):
+    """ Used to encode higher-dimension inputs into a
+    lower dimension latent space and decode latent vectors
+    back to the original feature space.
+
+    """
+
+    _TRANSFORMER_NAME = "sk_transformer.pkl"
+    _SCALER_NAME = "sk_scaler.pkl"
+    _METADATA_NAME = "metadata.yaml"
+
+    def __init__(
+        self,
+        transformer: TransformerMixin,
+        scaler: StandardScaler,
+        enforce_positive_outputs: bool = False,
+    ):
+        self.transformer = transformer
+        self.scaler = scaler
+        self.enforce_positive_outputs = enforce_positive_outputs
+
+    def predict(self, x: Sequence[np.ndarray]) -> Sequence[np.ndarray]:
```

Comment:
I think this could use some documentation.  I don't offhand know why we might want a rountrip method named "predict".  I assume it's just to adhere to the structure of an interface?

Reply:
Right, since this is going to be used in the same way as the dense autoencoder, I made its `predict` method behave similarly. This method is unlikely to actually get called in the actual reservoir experiments, it's main usefulness is when testing whether the autoencoder/transformer's error will negatively affect the prognostic run (e.g. the experiments I did last week). I'll add a comment explaining this.